### PR TITLE
tests for tsweb-76

### PR DESCRIPTION
**Pull request by YevheniiKel**

### CODE REVIEW

File: ui_tests_playwright/automation/test/ui/JobsBlock/JobsBlockNoContentful.test.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+	await driver
+		.locator(`[data-id*=${Containers.careerCardWithoutModifier}]`)
+		.nth(1)
+		.click();
+	await driver.getByTestId(Career.ApplyNowButton).click();
+
+	await driver.getByTestId(Career.Modal_firstNameInput).fill(' '); // One space
+	await driver
+		.getByTestId(Career.Modal_lastNameInput)
+		.fill('                                         '); // Many spaces
```
Comment (VladFedash) [inline ui_tests_playwright/automation/test/ui/JobsBlock/JobsBlockNoContentful.test.ts:24]: you can try to use smth like this
' '.repeat(n)

File: ui_tests_playwright/automation/test/ui/JobsBlock/JobsBlockNoContentful.test.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+	await baseDriverSteps.goToUrl(UrlProvider.careerUrl());
+});
+
+test('Check that "First Name" and "Last Name" input fields does not accept only spaces in "Apply for a Job" modal window on job page @Regression @JobsBlock @TSWEB-76', async () => {
+	await driver
+		.locator(`[data-id*=${Containers.careerCardWithoutModifier}]`)
+		.nth(1)
+		.click();
+	await driver.getByTestId(Career.ApplyNowButton).click();
+
```
Comment (VladFedash) [inline ui_tests_playwright/automation/test/ui/JobsBlock/JobsBlockNoContentful.test.ts:20]: lb

File: ui_tests_playwright/automation/test/ui/aboutUs/Careers_ContactUs.test.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+test.beforeEach(async () => {
+    await baseDriverSteps.createsNewBrowser();
+    await baseDriverSteps.goToUrl(UrlProvider.careerUrl());
+    await driver.getByTestId(Button.NavigationTab_ContactUs).click();
+});
+
+test("Check that 'Full Name' input field does not accept only spaces in the 'Contact Us' form @Regression @ContactUs @TSWEB-76", async () => {
+	const testData: Array<string> = [
+		" ", // One space
+		"                                         " // Many spaces
```
Comment (VladFedash) [inline ui_tests_playwright/automation/test/ui/aboutUs/Careers_ContactUs.test.ts:18]: same

**Review by VladFedash — CHANGES_REQUESTED** [review]

File: ui_tests_playwright/automation/test/ui/JobsBlock/JobsBlockNoContentful.test.ts (hunk 2 above)
Comment (YevheniiKel) [reply ui_tests_playwright/automation/test/ui/JobsBlock/JobsBlockNoContentful.test.ts:17]: it's different test blocks separation

Comment (VladFedash) [reply ui_tests_playwright/automation/test/ui/JobsBlock/JobsBlockNoContentful.test.ts:17]: But it's the same test, at least I don't see one more test() construction in the current place.

File: ui_tests_playwright/automation/test/ui/JobsBlock/JobsBlockNoContentful.test.ts [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+	await baseDriverSteps.createsNewBrowser();
+	await baseDriverSteps.goToUrl(UrlProvider.careerUrl());
+});
+
+test('Check that "First Name" and "Last Name" input fields does not accept only spaces in "Apply for a Job" modal window on job page @Regression @JobsBlock @TSWEB-76', async () => {
+	await driver.locator(`[data-id*=${Containers.careerCardWithoutModifier}]`).nth(1).click();
+	await driver.getByTestId(Career.ApplyNowButton).click();
+
+	await driver.getByTestId(Career.Modal_firstNameInput).fill(" ");
+	await driver.getByTestId(Career.Modal_lastNameInput).fill(" ".repeat(99));
```
Comment (VladFedash) [inline ui_tests_playwright/automation/test/ui/JobsBlock/JobsBlockNoContentful.test.ts:19]: Why did you choose the 99 magic number?)

**Review by VladFedash — APPROVED** [review]

**Review by VladFedash — APPROVED** [review]

File: ui_tests_playwright/automation/test/ui/aboutUs/Careers_ContactUs.test.ts [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+test("Check that 'Full Name' input field does not accept only spaces in the 'Contact Us' form @Regression @ContactUs @TSWEB-76", async () => {
+	const testData: Array<string> = [
+		" ", 
+		" ".repeat(149) // Field accepts up to 149 characters
+	];
+
+    for(const data of testData){
+		await driver.getByTestId(ContactUs.fullNameInput).fill(data);
+		await driver.getByTestId(ContactUs.sendRequestButton).click();
+		const actualErrorText_FullName = await driver.getByTestId(ContactUs.fullNameInput).locator(Input.fieldErrorSelector).textContent();
```
Comment (anastasiia-liashova) [inline ui_tests_playwright/automation/test/ui/aboutUs/Careers_ContactUs.test.ts:24]: `const actualErrorText_FullName = driver.getByTestId(ContactUs.fullNameInput).locator(Input.fieldErrorSelector);
await expect(actualErrorText_FullName ).toHaveText('Please enter your name');`

File: ui_tests_playwright/automation/test/ui/JobsBlock/JobsBlockNoContentful.test.ts [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+import Containers from '../../../identifiers/Containers';
+import Input from '../../../identifiers/Input';
+
+test.beforeEach(async () => {
+	await baseDriverSteps.createsNewBrowser();
+	await baseDriverSteps.goToUrl(UrlProvider.careerUrl());
+});
+
+test('Check that "First Name" and "Last Name" input fields does not accept only spaces in "Apply for a Job" modal window on job page @Regression @JobsBlock @TSWEB-76', async () => {
+	await driver.locator(`[data-id*=${Containers.careerCardWithoutModifier}]`).nth(1).click();
```
Comment (anastasiia-liashova) [inline ui_tests_playwright/automation/test/ui/JobsBlock/JobsBlockNoContentful.test.ts:15]: await driver.getByTestId(/CardWrapper/).click();

File: ui_tests_playwright/automation/test/ui/JobsBlock/JobsBlockNoContentful.test.ts [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+	await driver.getByTestId(Career.ApplyNowButton).click();
+
+	await driver.getByTestId(Career.Modal_firstNameInput).fill(" ");
+	await driver.getByTestId(Career.Modal_lastNameInput).fill(" ".repeat(99)); // Fieдd accepts up to 100 characters
+	await driver.getByTestId(Career.Modal_sendRequestButton).click();
+
+	const actualErrorText_FirstName = await driver
```
Comment (anastasiia-liashova) [inline ui_tests_playwright/automation/test/ui/JobsBlock/JobsBlockNoContentful.test.ts:22]: `const actualErrorText_FirstName = driver.getByTestId(Career.Modal_firstNameInput).locator(Input.fieldErrorSelector);`

File: ui_tests_playwright/automation/test/ui/JobsBlock/JobsBlockNoContentful.test.ts [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+		.getByTestId(Career.Modal_firstNameInput)
+		.locator(Input.fieldErrorSelector)
+		.textContent();
+	const actualErrorText_LastName = await driver
+		.getByTestId(Career.Modal_lastNameInput)
+		.locator(Input.fieldErrorSelector)
+		.textContent();
+	expect(actualErrorText_FirstName).toEqual('Please enter your name');
```
Comment (anastasiia-liashova) [inline ui_tests_playwright/automation/test/ui/JobsBlock/JobsBlockNoContentful.test.ts:30]: `await expect(actualErrorText_FirstName).toHaveText('Please enter your name');`

File: ui_tests_playwright/automation/test/ui/JobsBlock/JobsBlockNoContentful.test.ts [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+
+	await driver.getByTestId(Career.Modal_firstNameInput).fill(" ");
+	await driver.getByTestId(Career.Modal_lastNameInput).fill(" ".repeat(99)); // Fieдd accepts up to 100 characters
+	await driver.getByTestId(Career.Modal_sendRequestButton).click();
+
+	const actualErrorText_FirstName = await driver
+		.getByTestId(Career.Modal_firstNameInput)
+		.locator(Input.fieldErrorSelector)
+		.textContent();
+	const actualErrorText_LastName = await driver
```
Comment (anastasiia-liashova) [inline ui_tests_playwright/automation/test/ui/JobsBlock/JobsBlockNoContentful.test.ts:26]: same

File: ui_tests_playwright/automation/test/ui/JobsBlock/JobsBlockNoContentful.test.ts [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+		.getByTestId(Career.Modal_lastNameInput)
+		.locator(Input.fieldErrorSelector)
+		.textContent();
+	expect(actualErrorText_FirstName).toEqual('Please enter your name');
+	expect(actualErrorText_LastName).toEqual('Please enter your last name');
```
Comment (anastasiia-liashova) [inline ui_tests_playwright/automation/test/ui/JobsBlock/JobsBlockNoContentful.test.ts:31]: same

**Review by anastasiia-liashova — CHANGES_REQUESTED** [review]

File: ui_tests_playwright/automation/identifiers/Career.ts [hunk 11]
```diff
@@ -8,4 +8,9 @@ export default class Career {
 	static jobHeaderAboutUsTabUa = 'NavigationTab-ПроКомпанію';
 	static jobHeaderReviewsTabUa = 'NavigationTab-Відгуки';
 	static jobHeaderContactUsTabUa = 'NavigationTab-ЗвязатисьЗНами';
+	// Apply for a Job modal window
+	static ApplyNowButton = "ApplyNowButton-SharedApplyNow";
```
Comment (anastasiia-liashova) [inline ui_tests_playwright/automation/identifiers/Career.ts:12]: Use 'ApplyNow' button from Button class

File: ui_tests_playwright/automation/identifiers/Career.ts [hunk 12]
```diff
@@ -8,4 +8,9 @@ export default class Career {
...
+	static Modal_firstNameInput = 'CareerInput-Name';
+	static Modal_lastNameInput = 'CareerInput-LastName';
```
Comment (anastasiia-liashova) [inline ui_tests_playwright/automation/identifiers/Career.ts:14]: same

File: ui_tests_playwright/automation/identifiers/Career.ts [hunk 13]
```diff
@@ -8,4 +8,9 @@ export default class Career {
...
+	static Modal_sendRequestButton = 'SubmitButton-SharedSendMessage';
```
Comment (anastasiia-liashova) [inline ui_tests_playwright/automation/identifiers/Career.ts:15]: SendButton form Button class

File: ui_tests_playwright/automation/identifiers/ContactUs.ts [hunk 14]
```diff
@@ -0,0 +1,4 @@
+export default class ContactUs {
+	static fullNameInput = 'ContactInput-FullName';
```
Comment (anastasiia-liashova) [inline ui_tests_playwright/automation/identifiers/ContactUs.ts:2]: same

File: ui_tests_playwright/automation/test/ui/JobsBlock/JobsBlockNoContentful.test.ts [hunk 15]
```diff
@@ -0,0 +1,34 @@
...
+
+test('Check that "First Name" and "Last Name" input fields does not accept only spaces in "Apply for a Job" modal window on job page @Regression @JobsBlock @TSWEB-76', async () => {
+	await driver.getByTestId(/CardWrapper/).click();
+	await driver.getByTestId(Career.ApplyNowButton).click();
+
+	await driver.getByTestId(Career.Modal_firstNameInput).fill(" ");
+	await driver.getByTestId(Career.Modal_lastNameInput).fill(" ".repeat(99)); // Field accepts up to 100 characters
+	await driver.getByTestId(Career.Modal_sendRequestButton).click();
+
+	const actualErrorText_FirstName = await driver
```
Comment (anastasiia-liashova) [inline ui_tests_playwright/automation/test/ui/JobsBlock/JobsBlockNoContentful.test.ts:22]: without await

**Review by anastasiia-liashova — CHANGES_REQUESTED** [review]

**Review by anastasiia-liashova — APPROVED** [review]

**Review by anastasiia-liashova — APPROVED** [review]

File: ui_tests_playwright/automation/test/ui/JobsBlock/JobsBlock.test.ts [hunk 16]
```diff
@@ -11,6 +11,7 @@ import ContainerByClass from '../../../components/Container/ContainerByClass';
 import Containers from '../../../identifiers/Containers';
 import JobPagePreconditions from '../../../preconditionsData/uiPreconditions/JobPagePreconditions';
 import {sessionValue} from '../../../runtimeVariables/SessionValue';
+import Input from '../../../identifiers/Input';
```
Comment (DaniilVostrikov) [inline ui_tests_playwright/automation/test/ui/JobsBlock/JobsBlock.test.ts:14]: looks like it is not used in the current file

**Review by DaniilVostrikov — CHANGES_REQUESTED** [review]

**Review by DaniilVostrikov — APPROVED** [review]

**Pull request merged**